src/amber/router/params.cr

```diff
@@ -1,20 +1,14 @@
 module Amber::Router
-  class Params < Hash(String, String)
-    alias KeyType = String | Symbol
-    property files = {} of String => Amber::Router::Files::File
+  struct Params
+    property files = {} of String => File
+    property store : HTTP::Params = HTTP::Params.new(Hash(String, Array(String)).new)
 
-    def json(key : KeyType)
-      JSON.parse(self[key]?.to_s)
+    forward_missing_to @store
```

Comment:
There was some stuff we had talked about here that I was going to do later. I'll do that as a different PR after this is merged.